functions/__sf_section_package.fish

```diff
@@ -22,13 +23,39 @@ function __sf_section_package -d "Display the local package version"
 
 	[ $SPACEFISH_PACKAGE_SHOW = false ]; and return
 
-	# Show package version only when repository is a package
-	[ -f ./package.json ]; or return
-	# Show package version only if npm is installed
-	type -q npm; or return
+	# Exit if there is no package.json or Cargo.toml
+	if not test -e ./package.json; and not test -e ./Cargo.toml
+		return
+	end
+
+	set -l package_version
+
+	# Check if package.json exists AND npm exists locally while supressing output to just exit code (-q)
+	if test -f ./package.json; and type -q npm
```

Comment:
I would suggest doing the `type -q npm` check before the file check for a sooner return for non-npm users.